src/main/java/pl/edu/amu/szi/forklift/objects/Shelf.java

```diff
@@ -2,8 +2,65 @@
 
 import javafx.scene.canvas.GraphicsContext;
 
+import java.util.ArrayList;
+import java.util.Objects;
+
 public class Shelf extends GameObject {
-    public Shelf(GraphicsContext gc, String imageSrc, float tileWidth, float tileHeight, int initX, int initY) {
+    private ArrayList<Package> packageList = new ArrayList<>();;
+    private int shelfMaxWeight; // max weight of pkg on shelf
+
+    public Shelf(GraphicsContext gc, String imageSrc, float tileWidth, float tileHeight, int initX, int initY, int shelfMaxWeight) {
         super(gc, imageSrc, tileWidth, tileHeight, initX, initY);
+        this.shelfMaxWeight = shelfMaxWeight;
```

Comment:
jeśli mamy tylko jeden typ półek, to wywaliłbym to z konstruktora do zmiennej statycznej (public static final int SHELF_MAX_WEIGHT = 25;)